app/controllers/match_contacts_controller.rb

```diff
@@ -22,72 +22,121 @@ def edit
   def update
     update_params = match_contacts_params
     saved = @match_contacts.update(update_params)
-    unless request.xhr?
-      if saved
-        flash[:notice] = "Match Contacts updated"
-        redirect_to match_path(@match)
-      else
-        raise @match_contacts.errors.full_messages.inspect
-        flash[:error] = "Please review the form problems below."
-        redirect_to :edit
-      end
+    return if request.xhr?
+
+    if saved
+      flash[:notice] = 'Match Contacts updated'
+      redirect_to match_path(@match)
+    else
+      raise @match_contacts.errors.full_messages.inspect
+      flash[:error] = 'Please review the form problems below.' # rubocop:disable Lint/UnreachableCode
+      redirect_to :edit
     end
   end
 
+  def reorder
+    return unless params[:join_contact]
+
+    order_number = 1 # Default to setting the contact as primary
+    order_number = params[:order_number].to_i if params[:order_number]
+    set_contact_order(params[:join_contact], order_number)
+    set_match
+    set_match_contacts
+    # byebug
+    return if request.xhr?
+
+    redirect_to match_path(@match)
+  end
+
   private
 
-    def match_scope
-      ClientOpportunityMatch.all
+  def set_contact_order(update_contact, order_number = 1)
+    join_contact = match.client_opportunity_match_contacts.find(update_contact)
+    contact_type = nil
+    match_contact_methods.each do |type, _|
+      contact_type = type if join_contact.send(type)
     end
 
-    def set_match
-      @match = match_scope.find params[:match_id]
+    # We currently only want one contact per order number, so strip order status for any other contacts in the group
+    a_t = ClientOpportunityMatchContact.arel_table
+    contacts = @match.client_opportunity_match_contacts.where(a_t[contact_type].eq(true)).where(contact_order: order_number)
```

Comment:
Do you need to drop into Arel here?
This looks like it could be:
```
contacts = @match.client_opportunity_match_contacts.where(contact_type: true, contact_order: order_number)
```

Reply:
This whole section should probably be wrapped in a transaction so that if some part of it fails, it will be atomic.
```
ClientOpportunityMatchContact.transaction do
...
end
```

Also, rather than pull back the contacts and then save them each one at a time:
```
@match.client_opportunity_match_contacts.where(a_t[contact_type].eq(true)).where(contact_order: order_number).update_all(contact_order: nil)
```
should accomplish that in one query, unless there are callbacks that need to be run.

Reply:
The update_all and transaction definitely make sense. I'll ge them included. 

As for the arel, I was running into an issue getting the AR query to work with the column name being variable (contact_type) so used arel. Happy to switch that up if there's a better way to do it.